.github/scripts/hot_fix_pr.sh

```diff
@@ -0,0 +1,291 @@
+            print_help
+            exit
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *) echo "Error: Unexpected option: $1" >&2
+            exit 1
+    esac
+done
+
+# Validate required parameters
+if [ -z "${PR_NUMBER}" ]; then
+    echo -e "Error: missing required parameter --pr-number\n"
+    print_help
+    exit 1
+fi
+
+if [ -z "${GITHUB_TOKEN}" ]; then
+    echo -e "Error: missing 'GITHUB_TOKEN' environment variable\n"
+    print_help
+    exit 1
+fi
+
+# Function to make GitHub API calls
+github_api() {
+    endpoint="$1"
+    method="${2:-GET}"
+    data="${3:-}"
+    
+    if [ -n "$data" ]; then
+        curl -s -H "Authorization: token $token" \
+             -H "Accept: application/vnd.github.v3+json" \
+             -X "$method" \
+             -d "$data" \
+             "https://api.github.com/repos/$ORG/$REPO/$endpoint"
+    else
+        curl -s -H "Authorization: token $token" \
+             -H "Accept: application/vnd.github.v3+json" \
+             -X "$method" \
+             "https://api.github.com/repos/$ORG/$REPO/$endpoint"
+    fi
+}
+
+# Function to comment on a PR
+comment_on_pr() {
+    pr_number="$1"
+    comment_body="$2"
+    
+    comment_data=$(jq -n --arg body "$comment_body" '{body: $body}')
+    echo "Adding comment to PR #$pr_number..."
+    github_api "issues/$pr_number/comments" "POST" "$comment_data" > /dev/null
+}
+
+# Function to exit with error and optional PR comment
+exit_with_error() {
+    error_message="$1"
+    exit_code="${2:-1}"
+    
+    echo "$error_message"
+    
+    if [ "${IS_COMMENT_TRIGGER}" = "true" ]; then
+        comment_on_pr "$PR_NUMBER" "$error_message"
+    fi
+    
+    exit "$exit_code"
+}
+
+# Function to create a PR
+create_pr() {
+    source_branch="$1"
+    target_branch="$2"
+    title="$3"
+    body="$4"
+    
+    pr_data=$(cat <<EOF
+{
+    "title": "$title",
+    "body": "$body",
+    "head": "$source_branch",
+    "base": "$target_branch"
+}
+EOF
+)
+    
+    echo "Creating PR from $source_branch to $target_branch..."
+    if [ "${DRY_RUN}" != "true" ]; then
+        result=$(github_api "pulls" "POST" "$pr_data")
+        pr_url=$(jq -r '.html_url // empty' <<< "$result")
+        
+        if [ -n "$pr_url" ]; then
+            echo "Created PR: $pr_url"
+            # Set global variables
+            if [ "$target_branch" = "staging" ]; then
+                STAGING_PR_URL="$pr_url"
+            elif [ "$target_branch" = "production" ]; then
+                PRODUCTION_PR_URL="$pr_url"
+            fi
+            echo "$pr_url"  # Return the URL
+        else
+            echo "Failed to create PR"
+            echo jq -r '.message // "Unknown error"' <<< "$result"
+            return 1
+        fi
+    fi
+}
+
+echo "Fetching PR #$PR_NUMBER information..."
+
+# Get PR information
+PR_INFO=$(github_api "pulls/$PR_NUMBER")
+
+# Check if PR exists
+if jq -e '.message' <<< "$PR_INFO" > /dev/null 2>&1; then
```

Comment:
I'm not sure about this one. If `.message` is there, that means success, no? So it should be `if ! jq`, no? 

Also, the api call could fail for any number of reasons - not just because the PR is not found. If there is an error in the api response, we will not see what it is. So maybe print the api output before failing?